scripts/lib/CIME/case/case_st_archive.py

```diff
@@ -652,22 +637,24 @@ def restore_from_archive(self, rest_dir=None, dout_s_root=None, rundir=None):
         dout_s_root = self.get_value("DOUT_S_ROOT")
     if rundir is None:
         rundir = self.get_value("RUNDIR")
-    if rest_dir is not None:
+    if rest_dir:
         if not os.path.isabs(rest_dir):
             rest_dir = os.path.join(dout_s_root, "rest", rest_dir)
     else:
-        rest_dir = os.path.join(dout_s_root, "rest", ls_sorted_by_mtime(os.path.join(dout_s_root, "rest"))[-1])
-
-    logger.info("Restoring restart from {}".format(rest_dir))
+        rest_root = os.path.join(dout_s_root, "rest")
+        if os.path.exists(rest_root):
+            rest_dir = os.path.join(rest_root, ls_sorted_by_mtime(os.path.join(dout_s_root, "rest"))[-1])
+    if rest_dir:
```

Comment:
Can you explain this change? It looks like it is allowing for the possibility that the restart directory doesn't exist... but naively, I would have thought that that should be an error.

Reply:
I don't think that there is a change here, just reordered some code.

Reply:
I mean adding the conditional `if rest_dir` around the block of code below. You have changed some of the above code so that there is now a possibility that `rest_dir` won't be set: if you enter the `else` block just above this, but then `if os.path.exists(rest_root)` is False. My sense is that this function should abort with an error message in the case that `rest_dir` remains unset at this point, rather than silently doing nothing (which is the case with your changes, I think). If there's a reason that it's okay for `rest_dir` to remain unset and for this to end up doing no copies, can you describe what that scenario would be?